src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+
+namespace opossum {
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
+ public:
+  virtual ~BaseFilter() = default;
+
+  virtual bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const = 0;
+};
+
+class ChunkColumnStatistics {
+ public:
+  static std::shared_ptr<ChunkColumnStatistics> build_statistics(DataType data_type, std::shared_ptr<BaseColumn> column);
+
+  void add_filter(std::shared_ptr<BaseFilter> filter) {
```

Comment:
@Bouncner - I am not too happy with the use of "filter" here. `add_filter` sounds like you are filtering something. What happens here is that you add the data structure that enables filter later on. Is "filter" the commonly used name for this?

Reply:
How else would you call something that filters?

Reply:
I couldn't come up with a better name or I would have mentioned it. Just wanted to bring this up.

Reply:
It's a matter of taste. I see your point with that naming clash. SAP uses both `min max filters` as well as `min max statistics`. But the idea of our data structures here are similar to well known data structures such as "Bloom filters" or `Adaptive Range Filters` (http://www.vldb.org/pvldb/vol6/p1714-kossmann.pdf).
So I think `filter` is fine in general.

Here, I think it might be confusing but I wouldn't call the other upcoming data structures `statistics` either. So, I have not idea for a better term.

Reply:
Ok, let's stay with "filter".